collab-database/tests/database_test/view_test.rs

```diff
@@ -190,33 +194,42 @@ async fn delete_inline_database_view_test() {
   }
 
   let views = database_test.views.get_all_views();
-  let view_id = views[1].id.clone();
-  assert_eq!(views.len(), 3);
+  assert_eq!(views.len(), 5);
 
-  database_test.views.delete_view(&view_id);
+  let deleted_view_id = "v3".to_string();
+  database_test.views.delete_view(&deleted_view_id);
   let views = database_test
     .views
     .get_all_views()
     .iter()
     .map(|view| view.id.clone())
     .collect::<Vec<String>>();
-  assert_eq!(views.len(), 2);
-  assert!(!views.contains(&view_id));
+  assert_eq!(views.len(), 4);
+  assert!(!views.contains(&deleted_view_id));
 }
 
 #[tokio::test]
 async fn duplicate_database_view_test() {
   let database_test = create_database_with_default_data(1, "1").await;
-  database_test.duplicate_linked_view("v1");
 
   let views = database_test.views.get_all_views();
   assert_eq!(views.len(), 2);
+
+  let view = database_test.get_view("v1").unwrap();
+  let duplicated_view = database_test.duplicate_linked_view("v1").unwrap();
+
+  let views = database_test.views.get_all_views();
+  assert_eq!(views.len(), 3);
+
+  assert_eq!(duplicated_view.name, format!("{}-copy", view.name));
+  assert_ne!(view.id, duplicated_view.id);
+  // modified and created time should also be different but the test completes within one second.
 }
 
 #[tokio::test]
 async fn duplicate_database_data_serde_test() {
   let database_test = create_database_with_default_data(1, "1").await;
-  let duplicated_database = database_test.duplicate_database();
```

Comment:
duplicate_database is different from `get_dall_database_data` because duplicate need to modify the ids in database. these ids includes row id.

Reply:
I've changed the duplication logic a bit. The `get_all_database_data` fetches all the database data (fields, row, views, etc) and places them into a `DatabaseData`. Then, if duplicating, use the `to_create_database_params` with `regenerate = true` to generate new view and row ids.

Reply:
to_create_database_params is kind of vague. Only need to regenerate the ids when duplicating a  database